doc/source/ScriptingAPI/DictionarySyntax/index.md

```diff
@@ -0,0 +1,65 @@
+# Dictionary Syntax #
+
+Gaffer scripts use Python's dictionary syntax and notation to define the node graph. As you add to and manipulate the node graph, the dictionary builds a compendium of instances of node and plug classes. Each node is a dictionary key, with a node class as its value. Each plug is a sub-key of a node's key, with a plug class as its value. The following pseudocode shows a rough approximation of a node graph's dictionary:
+
+```python
+script = {
+    ["Node1"]:Module.NodeType1()
+    ["Node1"]["Plug1"]:Module.PlugType1()
+    ["Node1"]["Plug2"]:Module.PlugType2()
+
+    ["Node2"]:Module.NodeType2()
+    ["Node2"]["Plug1"]:Module.PlugType1()
+    ["Node2"]["in"]:GafferScene.ScenePlug( ["Node1"]["out"] )
+    
+    ["Node3"]:Module.NodeType3()
+    ["Node3"]["in"]:GafferScene.ScenePlug( ["Node2"]["out"] )
+}
+```
+
+When the script is evaluated, Gaffer traverses the dictionary, using the _in_ plugs of nodes to build a parent-child hierarchy. This becomes a dependency graph, as represented in the _Graph Editor_. At the start of all Gaffer node graphs is a root ScriptNode object, referenced by the `script` variable.
+
+
+## Adding Keys to Gaffer Dictionaries ##
+
+In Gaffer, adding nodes to the node graph is synonymous with adding keys to the `script` variable.
+
+
+### Python syntax ###
+
+```python
+script["newNode"] = Gaffer.Node
+```
+
+> Caution :
+> Adding nodes and plugs using Python dictionary syntax risks overwriting an existing node or plug.
+
+If you are familiar with Python, you probably know the syntax for adding keys to a dictionary. Gaffer's `script` variable is compatible with it, but it is not the best practice.
```

Comment:
This value judgement feels a bit weird.  You can use the dictionary syntax for adding nodes if you want to overwrite existing nodes, or the addChild syntax if you want want to Gaffer to automatically find a unique name.  We could just as easily say "Caution : Using addChild risks having your node renamed, which might not be what you intended".

Reply:
I could probably back off from that prescription in this section, but in other parts I recommend against it. Are there legitimate use cases where it's preferable to use the more dangerous method?

Reply:
Looking at this again, this is fine as written.  Not sure why it felt weird when I first read it.